src/connector/src/parser/debezium/mongo_json_parser.rs

```diff
@@ -0,0 +1,278 @@
+        ))
+    })?;
+    let id_field = bson_doc
+        .get("_id")
+        .ok_or_else(|| {
+            RwError::from(ProtocolError(
+                "Debezuim Mongo requires document has a `_id` field".into(),
+            ))
+        })?
+        .clone();
+    let id: Datum = match id_type {
+        DataType::Jsonb => ScalarImpl::Jsonb(id_field.into()).into(),
+        DataType::Varchar => match id_field {
+            serde_json::Value::String(s) => Some(ScalarImpl::Utf8(s.into())),
+            serde_json::Value::Object(obj) if obj.contains_key("$oid") => Some(ScalarImpl::Utf8(
+                obj["$oid"].as_str().to_owned().unwrap_or_default().into(),
+            )),
+            _ => Err(RwError::from(ProtocolError(format!(
+                "Can not convert bson {:?} to {:?}",
+                id_field,
+                id_type
+            ))))?,
+        },
+        DataType::Int32 => {
+            if let serde_json::Value::Object(ref obj) = id_field  && obj.contains_key("$numberInt"){
+                    let int_str = obj["$numberInt"].as_str().unwrap_or_default();
+                    Some(ScalarImpl::Int32(int_str.parse().unwrap_or_default()))
+                } else {
+                    Err(RwError::from(ProtocolError(format!(
+                        "Can not convert bson {:?} to {:?}",
+                        id_field,
+                        id_type
+                    ))))?
+            }
+        }
+        DataType::Int64 => {
+            if let serde_json::Value::Object(ref obj) = id_field && obj.contains_key("$numberLong") {
+                    let int_str = obj["$numberLong"].as_str().unwrap_or_default();
+                    Some(ScalarImpl::Int64(int_str.parse().unwrap_or_default()))
+                } else {
+                    Err(RwError::from(ProtocolError(format!(
+                        "Can not convert bson {:?} to {:?}",
+                        id_field,
+                        id_type
+                    ))))?
+            }
+        }
+        _ => unreachable!("DebeziumMongoJsonParser::new must ensure _id column datatypes."),
+    };
+    let payload: Datum = match payload_type {
+        DataType::Jsonb => ScalarImpl::Jsonb(bson_value.into()).into(),
+        DataType::Varchar => ScalarImpl::Utf8(bson_str.into()).into(),
+        _ => unreachable!("DebeziumMongoJsonParser::new must ensure payload column datatypes."),
+    };
+
+    Ok((id, payload))
+}
+impl DebeziumMongoJsonParser {
+    pub fn new(rw_columns: Vec<SourceColumnDesc>, source_ctx: SourceContextRef) -> Result<Self> {
+        let id_column = rw_columns
+            .iter()
+            .find(|desc| {
+                desc.name == "_id"
+                    && matches!(
+                        desc.data_type,
+                        DataType::Jsonb
+                            | DataType::Varchar
+                            | DataType::Int32
+                            | DataType::Int64
```

Comment:
Any doc on key type limits?

Reply:
We could extend them if other _id type appears.